tidb/join/README.md

```diff
@@ -0,0 +1,53 @@
+## Introduction
+
+This is the homework for PingCAP Talent Plan Online of week 4. This homework is a simplified version of [ACM SIGMOD Programming Contest 2018](http://sigmod18contest.db.in.tum.de/index.shtml).
+
+The task is to evaluate batches of join queries on a set of pre-defined relations. Each join query specifies two relations, multiple (equality) join predicates, and one (sum) aggregation. The challenge is to fully utilize the CPU and memory resources and execute the queries as fast as possible.
+
+NOTE: **go 1.12 is required**
+
+## Details
+
+The simple interface `Join(f0, f1 string, offset0, offset1 []int) (sum uint64)` is defined in `join.go`. Our test harness will feed two relations and two columns' offsets array to the interface every time, and check the correctness of the output result. Explaination to the four input arguments and one output argument of the interface are list as follows:
+
+- **f0**: File name of the given relation 0.
+- **f1**: File name of the given relation 1.
+- **offset0**: Offsets of which columns the given relation 0 should be joined.
```

Comment:
how about:
```
- **cols0**: Column ordinals of the given relation0 to be joined.
- **cols1**: Column ordinals of the given relation2 to be joined.

The join predicate(equality) is constructed according to cols0 and cols1 in the form of "cols0[i] = cols1[i]"
```

Reply:
For my part, `offset` may be better?
I use `offset` here to leave the usage of `cols` to the coders.
Since the form of the join predicates are actually `cols0[offset0[i]] = cols1[offset1[i]]`.